### PR TITLE
Update template inheritance docs

### LINKED ISSUE #5924: Document Cascadable Themes
RFC: https://github.com/silverstripe/silverstripe-framework/issues/5604
Pull request: https://github.com/silverstripe/silverstripe-framework/pull/5804

Document how to configure theme cascades (through PHP and YAML), mainly by updating https://github.com/silverstripe/silverstripe-framework/blob/master/docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md

# PR:

- [ ] https://github.com/silverstripe/silverstripe-framework/pull/7532

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes #5924

### CODE REVIEW

File: docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md [hunk 1]
```diff
@@ -139,3 +80,97 @@ If your classes have in a namespace, the Layout folder will be a found inside of
...
+there may be multiple instances of any matching path for a template across the set of all themes, a cascading
+search is done in order to determine the resolved template for any specified string.
+
+In order to declare the priority for this search, themes can be declared in a cascading fashion in order
+to determine resolution priority. This search is based on the following three configuration values:
+
+ - `SilverStripe\View\SSViewer.themes` - The list of all themes in order of priority. This includes the default
+   set via `$default` as a theme set. This config is normally set by the web developer.
+ - `SilverStripe\Core\Manifest\ModuleManifest.module_priority` - The priority of modules within which $default
+   theme templates should be sorted. This config is normally set by the module author, and does not normally need
```
Comment (flamerohr) [inline docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: is this a list/map/dictionary of priorities?

File: docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md [hunk 2]
```diff
@@ -3,96 +3,37 @@ summary: Override and extend module and core markup templates from your applicat
...
-
-## Nested Layouts through `$Layout`
+### Nested Layouts through `$Layout` type
 
 SilverStripe has basic support for nested layouts through a fixed template variable named `$Layout`. It's used for 
 storing top level template information separate to individual page layouts.
 
 When `$Layout` is found within a root template file (one in `templates`), SilverStripe will attempt to fetch a child 
-template from the `templates/Layout` directory. It will do a full sweep of your modules, core and custom code as it 
+template from the `templates/<namespace>/Layout` directory. It will do a full sweep of your modules, core and custom code as it
```
[the remark this answers is not in the document]
Comment (tractorcow) [reply docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: Yes.

File: docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md [hunk 3]
```diff
@@ -3,96 +3,37 @@ summary: Override and extend module and core markup templates from your applicat
...
-As we've added a new file, make sure you flush your SilverStripe cache by visiting `http://yoursite.com/?flush=1`
-</div>
-
-Template inheritance works on more than email templates. All files within the `templates` directory including `includes`, 
-`layout` or anything else from core (or add-on's) template directory can be overridden by being located inside your 
-`mysite/templates` directory. SilverStripe keeps an eye on what templates have been overridden and the location of the
-correct template through a [ThemeResourceLoader](api:SilverStripe\View\ThemeResourceLoader).
+Typically all templates within one of the above locations will be nested in a folder deterministically through
+the fully qualified namespace of the underlying class, and an optional `type` specifier to segment template types.
+Basic template types include `Layout` and `Includes`, and a less commonly used `Content` type.
```
[the remark this answers is not in the document]
Comment (tractorcow) [reply docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: Not really, but these are, at a higher level, all examples of "typed" templates, so these all demonstrate the concept despite their functional differences.

Layout and Content are closer to one another, whereas Includes behaves differently.

File: docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md (hunk 1 above)
Comment (tractorcow) [reply docs/en/02_Developer_Guides/01_Templates/05_Template_Inheritance.md]: List.

**Comment by tractorcow** [comment]
OK updated @flamerohr :)

**Pull request merged**